### PR TITLE
Add Resources service with methods for device

### PR DESCRIPTION
**Pull request by krystyw**

### CODE REVIEW

File: resources.go [hunk 1]
```diff
@@ -0,0 +1,185 @@
...
+	obj := &DataResponse{}
+	err := s.getResourcesByEndpoint(obj, endpoint, filters...)
+	if err != nil {
+		return nil, nil, err
+	}
+	return obj.Items, &obj.ListParams, nil
+}
+
+// GetEventsByDevice requests from CloudThing device's events with set filters
+func (s *ResourcesServiceOp) GetEventsByDevice(device *Device, filters ...interface{}) ([]EventPoint, *ListParams, error) {
```
Comment (ukaszki) [inline resources.go]: może lepiej GetEventsByDeviceID

File: resources.go [hunk 2]
```diff
@@ -0,0 +1,185 @@
...
+	obj := &EventsResponse{}
+	err := s.getResourcesByEndpoint(obj, endpoint, filters...)
+	if err != nil {
+		return nil, nil, err
+	}
+	return obj.Items, &obj.ListParams, nil
+}
+
+// GetCommandsByDevice requests from CloudThing device's commands with set filters
+func (s *ResourcesServiceOp) GetCommandsByDevice(device *Device, filters ...interface{}) ([]CommandPoint, *ListParams, error) {
```
Comment (ukaszki) [inline resources.go]: jw

File: resources.go [hunk 3]
```diff
@@ -0,0 +1,185 @@
...
+	obj := &CommandsResponse{}
+	err := s.getResourcesByEndpoint(obj, endpoint, filters...)
+	if err != nil {
+		return nil, nil, err
+	}
+	return obj.Items, &obj.ListParams, nil
+}
+
+// WriteDataForDevice sends data to CloudThing device's for saving
+func (s *ResourcesServiceOp) WriteDataForDevice(device *Device, points []DataPoint) ([]DataPoint, error) {
```
Comment (ukaszki) [inline resources.go]: jw

File: resources.go [hunk 4]
```diff
@@ -0,0 +1,185 @@
...
+	obj := make([]DataPoint, 0)
+	err := s.writeResourcesByEndpoint(obj, points, endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// WriteEventsForDevice sends events to CloudThing device's for saving
+func (s *ResourcesServiceOp) WriteEventsForDevice(device *Device, points []EventPoint) ([]EventPoint, error) {
```
Comment (ukaszki) [inline resources.go]: jw

File: resources.go [hunk 5]
```diff
@@ -0,0 +1,185 @@
...
+	obj := make([]EventPoint, 0)
+	err := s.writeResourcesByEndpoint(obj, points, endpoint)
+	if err != nil {
+		return nil, err
+	}
+	return obj, nil
+}
+
+// WriteCommandsForDevice sends events to CloudThing device's for saving
+func (s *ResourcesServiceOp) WriteCommandsForDevice(device *Device, points []CommandPoint) ([]CommandPoint, error) {
```
Comment (ukaszki) [inline resources.go]: jw

**Review by ukaszki — CHANGES_REQUESTED** [review]

**Comment by krystyw** [comment]
Fixed

**Pull request merged**